crates/connector_proxy/src/interceptors/airbyte_source_interceptor.rs

```diff
@@ -0,0 +1,429 @@
+use crate::apis::{FlowCaptureOperation, InterceptorStream};
+
+use crate::errors::{create_custom_error, raise_custom_error, Error};
+use crate::libs::airbyte_catalog::{
+    self, ConfiguredCatalog, ConfiguredStream, DestinationSyncMode, Message, Range, ResourceSpec,
+    Status, SyncMode,
+};
+use crate::libs::command::send_sigcont;
+use crate::libs::json::{create_root_schema, tokenize_jsonpointer};
+use crate::libs::protobuf::{decode_message, encode_message};
+use crate::libs::stream::stream_all_airbyte_messages;
+
+use async_stream::try_stream;
+use bytes::Bytes;
+use protocol::capture::{
+    discover_response, validate_response, DiscoverRequest, DiscoverResponse, Documents,
+    PullRequest, PullResponse, SpecResponse, ValidateRequest, ValidateResponse,
+};
+use protocol::flow::{DriverCheckpoint, Slice};
+use std::collections::HashMap;
+use std::str::FromStr;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use validator::Validate;
+
+use futures_util::StreamExt;
+use json_pointer::JsonPointer;
+use serde_json::value::RawValue;
+use std::fs::File;
+use std::io::Write;
+use tempfile::{Builder, TempDir};
+use tokio_util::io::StreamReader;
+
+const CONFIG_FILE_NAME: &str = "config.json";
+const CATALOG_FILE_NAME: &str = "catalog.json";
+const STATE_FILE_NAME: &str = "state.json";
+
+pub struct AirbyteSourceInterceptor {
+    validate_request: Arc<Mutex<Option<ValidateRequest>>>,
+    stream_to_binding: Arc<Mutex<HashMap<String, usize>>>,
+    tmp_dir: TempDir,
+}
+
+impl AirbyteSourceInterceptor {
+    pub fn new() -> Self {
+        AirbyteSourceInterceptor {
+            validate_request: Arc::new(Mutex::new(None)),
+            stream_to_binding: Arc::new(Mutex::new(HashMap::new())),
+            tmp_dir: Builder::new()
+                .prefix("airbyte-source-")
+                .tempdir_in("/connector-tmpfs")
```

Comment:
I spent some time on the file location. The struggle here is that it is difficult to find a path that the connector could write to.
- The `tmp` dir was mounted to local, so not safe.
- The proxy is not running as a Root user, so does not have a permission to create a dir in most places in the file system. And the available locations could be image-dependent and complex.

On the other hand, when researching, I found the `tmpfs` is pretty easy to set up, with just a few lines of code change. So I added this as a placehoder. And if storing data as files in host machine memory is not working well for other coonsiderations, especially in k8s env, I'll look into more on this..

Reply:
Yes, `/var/tmp` works fine. Thank you! We can plan the next steps after this. Removing the `/tmp` binding would be ideal.